tensorboard/data/server/data_compat.rs

```diff
@@ -0,0 +1,200 @@
+                Some(SCALARS_PLUGIN_NAME) => {
+                    md.data_class = pb::DataClass::Scalar.into();
+                }
+                _ => {}
+            };
+            Box::new(md)
+        }
+        (None, _) => Box::new(pb::SummaryMetadata::default()),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn tensor_shape(dims: &[i64]) -> pb::TensorShapeProto {
+        pb::TensorShapeProto {
+            dim: dims
+                .iter()
+                .map(|n| pb::tensor_shape_proto::Dim {
+                    size: *n,
+                    ..Default::default()
+                })
+                .collect(),
+            ..Default::default()
+        }
+    }
+
+    mod scalars {
+        use super::*;
+
+        #[test]
+        fn test_tf1x_simple_value() {
+            let md = pb::SummaryMetadata {
+                plugin_data: Some(PluginData {
+                    plugin_name: "ignored_plugin".to_string(),
+                    content: b"ignored_content".to_vec(),
+                    ..Default::default()
+                }),
+                ..Default::default()
+            };
+            let v = Value::SimpleValue(0.125);
+            let result = initial_metadata(Some(md), &v);
+
+            assert_eq!(
+                *result,
+                pb::SummaryMetadata {
+                    plugin_data: Some(PluginData {
+                        plugin_name: SCALARS_PLUGIN_NAME.to_string(),
+                        ..Default::default()
+                    }),
+                    data_class: pb::DataClass::Scalar.into(),
+                    ..Default::default()
+                }
+            );
+        }
+
+        #[test]
+        fn test_tf2x_scalar_tensor_without_dataclass() {
+            let md = pb::SummaryMetadata {
+                plugin_data: Some(PluginData {
+                    plugin_name: SCALARS_PLUGIN_NAME.to_string(),
+                    content: b"preserved!".to_vec(),
+                    ..Default::default()
+                }),
+                ..Default::default()
+            };
+            let v = Value::Tensor(pb::TensorProto {
+                dtype: pb::DataType::DtFloat.into(),
+                tensor_shape: Some(tensor_shape(&[])),
+                float_val: vec![0.125],
+                ..Default::default()
+            });
+            let result = initial_metadata(Some(md), &v);
+
+            assert_eq!(
+                *result,
+                pb::SummaryMetadata {
+                    plugin_data: Some(PluginData {
+                        plugin_name: SCALARS_PLUGIN_NAME.to_string(),
+                        content: b"preserved!".to_vec(),
+                        ..Default::default()
+                    }),
+                    data_class: pb::DataClass::Scalar.into(),
+                    ..Default::default()
+                }
+            );
+        }
+    }
+
+    mod unknown {
```

Comment:
Yeah, I’m using them as such, since they show up in the output like this:

```
test data_compat::tests::scalars::test_tf1x_simple_value ... ok
test data_compat::tests::scalars::test_tf2x_scalar_tensor_without_dataclass ... ok
test data_compat::tests::unknown::test_custom_plugin_with_dataclass ... ok
test data_compat::tests::unknown::test_empty ... ok
test data_compat::tests::unknown::test_unknown_plugin_no_dataclass ... ok
```

I haven’t done this before and don’t have strong opinions about it, but
it seemed pretty reasonable, so I figured that I’d try it.